File: MPChartLib/src/main/java/com/github/mikephil/charting/formatter/DefaultFillFormatter.java
package com.github.mikephil.charting.formatter;


import com.github.mikephil.charting.data.LineData;
import com.github.mikephil.charting.interfaces.dataprovider.LineDataProvider;
import com.github.mikephil.charting.interfaces.datasets.ILineDataSet;

/**
 * Default formatter that calculates the position of the filled line.
 *
 * @author Philipp Jahoda
 */
public class DefaultFillFormatter implements IFillFormatter
{

    @Override
    public double getFillLinePosition(ILineDataSet dataSet, LineDataProvider dataProvider) {

        double fillMin = 0d;
        double chartMaxY = dataProvider.getYChartMax();
        double chartMinY = dataProvider.getYChartMin();

        LineData data = dataProvider.getLineData();

        if (dataSet.getYMax() > 0 && dataSet.getYMin() < 0) {
            fillMin = 0d;
        } else {

            double max, min;

            if (data.getYMax() > 0)
                max = 0d;
            else
                max = chartMaxY;
            if (data.getYMin() < 0)
                min = 0d;
            else
                min = chartMinY;

            fillMin = dataSet.getYMin() >= 0 ? min : max;
        }

        return fillMin;
    }
}

File: MPChartLib/src/main/java/com/github/mikephil/charting/formatter/DefaultFillFormatter.kt
package com.github.mikephil.charting.formatter

import com.github.mikephil.charting.interfaces.dataprovider.LineDataProvider
import com.github.mikephil.charting.interfaces.datasets.ILineDataSet

/**
 * Default formatter that calculates the position of the filled line.
 */
open class DefaultFillFormatter : IFillFormatter {

    override fun getFillLinePosition(dataSet: ILineDataSet?, dataProvider: LineDataProvider?): Float {
        val fillMin: Float
        val chartMaxY = dataProvider!!.yChartMax
        val chartMinY = dataProvider.yChartMin
        val data = dataProvider.lineData
        fillMin = if (dataSet!!.yMax > 0 && dataSet.yMin < 0) {
            0f
        } else {
            val max: Float = if (data.yMax > 0) 0f
            else
                chartMaxY
            val min: Float = if (data.yMin < 0)
                0f
            else
                chartMinY
            if (dataSet.yMin >= 0)
                min
            else
                max
        }
        return fillMin
    }
}
